src/NHibernate.Test/CfgTest/ConfigurationSerializationTests.cs

```diff
@@ -82,6 +84,18 @@ public void Basic_CRUD_should_work()
 				var p = session.Get<Parent>(parentId);
 				Assert.That(p, Is.Null);
 			}
+
+			if (((ISessionFactoryImplementor) sf).ConnectionProvider.Driver is FirebirdClientDriver fbDriver)
```

Comment:
Well why not. It would hide that Firebird specific trouble from other NHibernate users, and after all, an ORM is about abstracting the persisting layer as much as possible.